### ISSUE TITLE
Problem finding sources from source maps

### ISSUE THREAD

**Issue by xp500** [googleapis/cloud-debug-nodejs#530]
#### Environment details

  - OS: macOS High Sierra 10.13.6 
  - Node.js version: v8.10.0
  - npm version: 5.6.0
  - `@google-cloud/debug-agent` version: 3.0.0

#### Description
I'm having trouble getting the debugger to work locally on 3.0.0. The root problem is that I can't configure the paths correctly.

I have the following file structure:

`/repo_root/code/rest/src/company/file.ts`
`/repo_root/code/rest/build/company/file.js`
`/repo_root/code/rest/build/company/file.js.map`

The program's cwd is `/repo_root/code/rest`

I implemented a very naive pathResolver which returns the absolute path of the file.
```
(scriptPath: string, knownFiles: string[], resolved: string[]) => {
        return knownFiles.filter(f => f.endsWith(scriptPath));
}
```

When debugging the problem, I noticed that the method `SourceMapper.mappingInfo` was returning the line `-1` and that the source file passed to the source map consumer for `code/rest/src/company/rest.ts` was `../../code/rest/src/company/rest.ts`. That's because the source path is calculated as 
```
path.relative(path.dirname(entry.mapFile), inputPath).replace(/\\/g, '/')
```
Where `entry.mapFile` is `build/company/file.js` but inputPath is `code/rest/src/company/rest.ts`. The `mapFile` is supposed to be relative to the cwd (`repo_root/code/rest`) but the `inputPath` is supposed to be relative to the project root. However, sin both are relative paths, the source path ends up being wrong (`../../code/rest/src/company/rest.ts`, which I think is relative to the sourcemap file).

Replacing `/code/rest` with an empty string (i.e. `../../src/company/rest.ts`) in the source path makes everything work. The source map consumer entry even has an `_absoluteSources` property with value `[ '../../src/company/rest.ts' ]`, which si the correct path relative to the source map.

So, I'm not sure if this is some configuration I'm missing or a bug with how the source path is calculated

**Comment by DominicKramer** [comment]
This looks like a bug in the sourcemapper to me.  Thanks for reporting it.

**Comment by DominicKramer** [comment]
This issue has been resolved in version 3.0.1.
